### PR TITLE
Feature/customer mutation endpoints

### PR DESCRIPTION
**Pull request by jack-mccabe-defra**

### CODE REVIEW

File: app/graphql/types/customer/mutation.gql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+input AuthenticationInput {
+    referenceNumber: Int!
+    memorableDate: String
+    memorableEvent: String
+    memorablePlace: String
+}
+
+extend type Mutation {
+    customerAuthenticateUpdate(customerAuthenticateUpdate: AuthenticationInput!): AuthenticationQuestions
+}
```
Comment (csaltmountain) [inline app/graphql/types/customer/mutation.gql:10]: Update to match naming conventions: https://www.apollographql.com/docs/technotes/TN0002-schema-naming-conventions/#field-names

```suggestion
input UpdateCustomerAuthenticationQuestionsInput {
    referenceNumber: Int!
    memorableDate: String
    memorableEvent: String
    memorablePlace: String
}

extend type Mutation {
    updateCustomerAuthenticationQuestions(input: updateCustomerAuthenticationQuestions!): CustomerAuthenticationQuestions
}
```

File: app/graphql/types/customer/customer.gql [hunk 2]
```diff
@@ -13,18 +22,13 @@ type CustomerStatus {
 }
 
 type AuthenticationQuestions {
```
Comment (csaltmountain) [inline app/graphql/types/customer/customer.gql:24]: Maybe this should be `CustomerAuthenticationQuestions` to follow the same convention as `CustomerInfo` etc

File: app/graphql/server.js [hunk 3]
```diff
@@ -10,13 +10,19 @@ import { ApolloServerPluginLandingPageLocalDefault } from '@apollo/server/plugin
 const graphqlPath = dirname(fileURLToPath(import.meta.url))
 
 export const schema = makeExecutableSchema({
-  typeDefs: await loadFiles(join(graphqlPath, 'types')),
-  resolvers: await loadFiles(join(graphqlPath, 'resolvers')),
+  typeDefs: await loadFiles(join(graphqlPath, 'types'), {
+    recursive: true,
+    requireMethod: async path => import(pathToFileURL(path))
```
Comment (csaltmountain) [inline app/graphql/server.js:15]: Is this `requireMethod` needed? Shouldn't be necessary as loading async anyways

File: app/graphql/resolvers/customer/base.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { findCustomerByReferenceHandler } from '../../../core/customer/query.js'
+import { updateCustomerAuthenticateHandler } from '../../../core/customer/mutation.js'
+
+export const Query = {
+  customer: function (_, { referenceNumber }) {
```
Comment (csaltmountain) [inline app/graphql/resolvers/customer/base.js:5]:
```suggestion
  customer (_, { referenceNumber }) {
```

File: app/graphql/resolvers/customer/base.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    return findCustomerByReferenceHandler(referenceNumber)
+  }
+}
+
+export const Mutation = {
+  customerAuthenticateUpdate: function (_, { customerAuthenticateUpdate }) {
```
Comment (csaltmountain) [inline app/graphql/resolvers/customer/base.js:11]:
```suggestion
  customerAuthenticateUpdate (_, { customerAuthenticateUpdate }) {
```

File: app/graphql/types/customer/customer.gql (hunk 2 above)
Comment (jack-mccabe-defra) [reply app/graphql/types/customer/customer.gql]: updated

File: app/graphql/server.js (hunk 3 above)
Comment (jack-mccabe-defra) [reply app/graphql/server.js]: Seems to be required to resolve imports on windows, can be removed when using linux

**Review by csaltmountain — APPROVED** [review]
Looks good 👍

There're a few naming convention things that won't match with the schema linting I've added in my PR, but let's merge this first and then I'll resolve them in my branch.

**Pull request merged**